src/screens/NanoContract/NanoContractRegisterScreen.js

```diff
@@ -0,0 +1,310 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useState } from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  Image,
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { isEmpty } from 'lodash';
+import HathorHeader from '../../components/HathorHeader';
+import { CircleInfoIcon } from '../../components/Icons/CircleInfo.icon';
+import NewHathorButton from '../../components/NewHathorButton';
+import OfflineBar from '../../components/OfflineBar';
+import SimpleInput from '../../components/SimpleInput';
+import { TextLabel } from '../../components/TextLabel';
+import { TextValue } from '../../components/TextValue';
+import { COLORS } from '../../styles/themes';
+import { firstAddressRequest, nanoContractRegisterReady, nanoContractRegisterRequest } from '../../actions';
+import { feedbackSucceedText, hasFailed, hasSucceeded, isLoading } from './helper';
+import Spinner from '../../components/Spinner';
+import FeedbackModal from '../../components/FeedbackModal';
+import errorIcon from '../../assets/images/icErrorBig.png';
+import checkIcon from '../../assets/images/icCheckBig.png';
+import { FeedbackContent } from '../../components/FeedbackContent';
+
+/**
+ * Verifies if the invalidModel of the form has an error message.
+ */
+function hasError(invalidModel) {
+  return Object
+    .values({ ...invalidModel })
+    .reduce((_hasError, currValue) => _hasError || !isEmpty(currValue), false);
+}
+
+/**
+ * Validates the formModel, returning the invalidModel.
+ * If there is no error in the formModel, the invalidModel is returned empty.
+ */
+function validate(formModel) {
+  const invalidModel = {};
+
+  if (!formModel.ncId) {
+    invalidModel.ncId = t`Nano Contract ID is required.`;
+  }
+
+  return invalidModel;
+}
+
+export function NanoContractRegisterScreen({ navigation }) {
+  const dispatch = useDispatch();
+  const { address, error } = useSelector((state) => state.firstAddress);
+  const registerState = useSelector((state) => ({
+    registerStatus: state.nanoContract.registerStatus,
+    registerFailureMessage: state.nanoContract.registerFailureMessage,
+  }));
+
+  const [isClean, setClean] = useState(true);
+  const [formModel, setFormModel] = useState({
+    ncId: null,
+  });
+  const [invalidModel, setInvalidModel] = useState({
+    ncId: null,
+  });
+
+  /* @param {
+   *   'nodeUrl'|'explorerUrl'|'explorerServiceUrl'|'walletServiceUrl'|'walletServiceWsUrl'
+   * } name */
+  const handleInputChange = (name) => (value) => {
```

Comment:
Since this method is being passed down to components as a prop, you should use the `useCallback` hook to memoize it, example:

```
  const handleInputChange = useCallback(
    (name) => (value) => {
      if (isClean) {
        setClean(false);
      }

      // update invalid model
      const invalidModelCopy = { ...invalidModel };
      delete invalidModelCopy[name];
      setInvalidModel(invalidModelCopy);

      // update form model
      const form = {
        ...formModel,
        [name]: value,
      };
      setFormModel(form);

      // validate form model and update invalid model
      setInvalidModel(validate(form));
    },
    [isClean, invalidModel, formModel]
  );
```

this will prevent unnecessary re-renders on the child component when the NanoContractRegisterScreen is re-rendered but the dependencies don't change (isClean, invalidModel, formModel)

Reply:
Good point. Thank you!

Reply:
Done.